### ISSUE TITLE
Handling of non-standard `datadir`

### ISSUE THREAD

**Issue by eliasp** [saltstack-formulas/mysql-formula#88]
Right now, there's no handling at all for non-default locations of `datadir`.

- a fresh deployment doesn't create the `datadir`
- changing an existing deployment doesn't move the existing `datadir` to its new location (shutdown of mysql service, moving `datadir`, starting mysql service again)

There might be some difficulties involved in this, as the package management of some distributions adds mechanisms (postinst etc.) to the MySQL package which expects the default path (`/var/lib/mysql`) and otherwise might break the package installation/always re-create the datadir, etc.

**Comment by hgfischer** [comment]
Yes, this is annoying. The configuration exists, but it only creates the datadir for some distributions. For Ubuntu or Debian, for example, it doesn't.

**Comment by noelmcloughlin** [comment]
related issue copied from  #81 

> Non-standard 'datadir' needs AppArmor changes on Ubuntu

> When using a non-standard `datadir` on Ubuntu, changes to AppArmor are required, otherwise MySQL won't start anymore.
See also: https://blogs.oracle.com/jsmyth/entry/apparmor_and_mysql  Similar changes are most likely required for SELinux enabled systems.
